### PR TITLE
Remove Option<bool> where possible #3819

### PR DESCRIPTION
**Pull request by Freakazoid182**
Replacing `Option<bool>` with `bool` where possible without changing any behavior or breaking APIs.
As I mention on #3819, I think the good convention is to resolve to either `true` or `false` as soon as possible.

- Adding `#[serde(default)]` to struct fields for API objects. This is to have `bool` fields resolve to `false`, when no value is provided. This is only possible for most create, read and delete operations. This seems to be the earliest moment to resolve the `Option<bool>`.
- Replacing `Option<bool>` for most fields in Create db_schema forms. These mostly seemed to be `Option<bool>` to match with the API object.
- Update flows have been left untouched
- `CommentReplyQuery::show_bot_accounts` must default to `true`. I've left this one `Option<bool>`
- The `CreateSite` API object sounds like a create object, but it actually does an update. I've left this untouched too.

I am in doubt a bit on whether there are any Apub objects that can be updated. Although it seems that these objects are shared between Create and Update actions, which seems to me that these should keep the `Option<bool>` fields.

### CODE REVIEW

**Comment by dullbananas** [comment]
`show_bot_accounts` can be changed to `hide_bot_accounts`

**Comment by Freakazoid182** [comment]
> `show_bot_accounts` can be changed to `hide_bot_accounts`

Your comment made me notice `CommentReplyQuery` isn't actually a deserialized object. It's actually always set explicitly, so it has no need to be an `Option<bool>`. Thanks :)

The field is named `show_bot_accounts `, also in the `LocalUser` mode. Renaming it might be confusing, and doesn't really relate to the changes I want to make in this PR. I'll just keep it named `show_bot_accounts`

File: scripts/lint.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
-set -e
+# set -e
```
Comment (Nutomic) [inline scripts/lint.sh:2]: Leave this as is.

**Comment by Nutomic** [comment]
Looks good, it really simplifies the code. Tests are currently failing, run `./scripts/test.sh`.

I suppose the federation code would also work with `#[serde(default)]`. But the current code isnt causing such problems like in the api, so I dont see a reason to change it.

**Comment by Nutomic** [comment]
@dessalines Can you test the code generation for lemmy-js-client with these changes? I fear that it wont be able to tell which fields are optional or mandatory.

**Comment by Freakazoid182** [comment]
> Looks good, it really simplifies the code. Tests are currently failing, run `./scripts/test.sh`.
> 
> I suppose the federation code would also work with `#[serde(default)]`. But the current code isnt causing such problems like in the api, so I dont see a reason to change it.

Tests are failing because I overlooked the fact that for some `boolean` fields in the database the default is `TRUE`. I've updated some of the resource `InsertForm` structs making `Option<bool>` fields bool fields. This will default them `false` when not set excitability in the builder. An example is the `post` table's `local` field. It's good the tests are catching this.

What do you think we should do here? I think there are 3 options:
- Keep the `Option<bool>` type for fields in `InsertForm` structs. This will undo some of the cleaner code.
- Always make boolean fields default to `FALSE` in the database. Requires always setting `true` explicitly in code for inserts.
- Remove any defaults for the database on `boolean` fields. Requires always setting `false` or `true` explicitly for inserts.

> @dessalines Can you test the code generation for lemmy-js-client with these changes? I fear that it wont be able to tell which fields are optional or mandatory.

I have not looked into the lemmy-js-client code generation before. Apologies if these changes totally overlooked this.

**Comment by dessalines** [comment]
I'd only meant for this to change the **db comment and post list queries**, not the API. To me it doesn't make any sense to have a ton of Options in API get and post requests, and only have the bool ones be mandatory.

**Comment by Freakazoid182** [comment]
> Yep, the generated typescript (and kotlin) types will make them appear required, which they shouldn't be. Should be explicitly optional.

Thanks for checking. I clearly went a bit overboard here :)
It's probably better for me to close this PR. It's likely also a bit too impactful in hindsight, also with the possible required changes needed on the Postgres column defaults.

Happy to have a look at just the comment and post list queries. Unless you were already planning to pick that up?

**Comment by dessalines** [comment]
You can grab those for sure, i'll be out for a week.

And also I apologize for not being more clear in my issue, that's my bad.

**Pull request closed without merging**